packages/graphql-codegen-cli/src/codegen.ts

```diff
@@ -202,21 +202,29 @@ export async function executeCodegen(input: CodegenContext | Types.Config): Prom
             const outputConfig = generates[filename];
             const hasPreset = !!outputConfig.preset;
 
-            const title = hasPreset
-              ? `Generate to ${filename} (using EXPERIMENTAL preset "${outputConfig.preset}")`
-              : `Generate ${filename}`;
+            const title = `Generate to ${filename}`;
 
             return {
               title,
-              task: (_, subTask) => {
+              task: async (_, subTask) => {
                 let outputSchemaAst: GraphQLSchema;
                 let outputSchema: DocumentNode;
                 const outputFileTemplateConfig = outputConfig.config || {};
                 let outputDocuments: Types.DocumentFile[] = [];
                 const outputSpecificSchemas = normalizeInstanceOrArray<Types.Schema>(outputConfig.schema);
-                const outputSpecificDocuments = normalizeInstanceOrArray<Types.OperationDocument>(
-                  outputConfig.documents
-                );
+                let outputSpecificDocuments = normalizeInstanceOrArray<Types.OperationDocument>(outputConfig.documents);
+
+                const preset: Types.OutputPreset | null = hasPreset
+                  ? typeof outputConfig.preset === 'string'
+                    ? await getPresetByName(outputConfig.preset, makeDefaultLoader(context.cwd))
+                    : outputConfig.preset
+                  : null;
+
+                if (preset) {
+                  if (preset.prepareDocuments) {
```

Comment:
That's the new feature we discussed yesterday, right? where the plugin can modify the `document` section

Reply:
yes!